src/app/pages/common/entity/table/expandable-table/expandable-table.component.ts

```diff
@@ -39,22 +52,36 @@ export class ExpandableTableComponent {
 
     this.tableConf.afterGetDataExpandable = (data: any) => {
       this.isEmpty = !data.length;
-      this.disabled = (this.isEmpty && this.tableConf.collapsedIfEmpty)
-        || this.tableConf.alwaysCollapsed
-        || this.tableConf.alwaysExpanded
-        || (!this.isEmpty && this.tableConf.expandedIfNotEmpty);
+      this.disabled = true;
       if (this.tableConf.limitRows) {
         return data.splice(0, this.tableConf.limitRows);
       }
       return data;
     };
   }
 
-  shouldBeCollapsed(): boolean {
-    return this.tableConf.alwaysCollapsed || (this.isEmpty && this.tableConf.collapsedIfEmpty);
+  ngAfterViewChecked(): void {
+    if (this.isExpanded) {
+      const tableHeader = this.appTable.nativeElement.querySelector('thead');
+      const detailsRow = this.appTable.nativeElement.querySelector('#actions-row');
+      const expandableHeader = this.appTable.nativeElement.querySelector('mat-expansion-panel-header');
+      const tableHeaderHeight = tableHeader ? tableHeader.offsetHeight : 0;
+      const expandableHeaderHeight = expandableHeader ? expandableHeader.offsetHeight : 0;
+      const detailsFooterHeight = detailsRow ? detailsRow.offsetHeight : 0;
+      const totalHeight = this.appTable.nativeElement.offsetHeight;
+      const maxRowsHeight = totalHeight - expandableHeaderHeight - tableHeaderHeight - detailsFooterHeight;
+      if (this.tableConf.limitRowsByMaxHeight) {
+        const prevRowsLimit = this.tableConf.limitRows;
+        const TABLE_ROW_SIZE = 48;
```

Comment:
All caps doesn't match [the code style](https://github.com/airbnb/javascript#naming--uppercase) in this case.